python/plugins/processing/algs/gdal/ogr2ogrdissolve.py

```diff
@@ -95,8 +99,9 @@ def getConsoleCommands(self):
         fields = self.getParameterValue(self.FIELDS)
         querystart = '-dialect sqlite -sql "SELECT ST_Union(' + geometry + ')'
         queryend = ' FROM ' + layername + ' GROUP BY ' + field + '"'
+        fieldList = getLayerFieldList(inLayer)
         if fields:
-            queryfields = ",*"
+            queryfields = ',%s' % fieldList
```

Comment:
Sidenote, it's advisable to switch to the newer style syntax `',{}'.format(fieldList)` ;)